src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -30,15 +40,37 @@ protected override int GetFields<T>(T instance, LazyStringValue key, object docu
                 newFields++;
             }
 
+            IndexField allFields = null;
+            _fields.TryGetValue(Constants.Documents.Indexing.Fields.AllFields, out allFields);
+            var AllFieldsOptions = allFields?.ToIndexFieldOptions();
             foreach ((var property, var propertyDescriptor) in documentToProcess.GetOwnProperties())
             {
+                var actualValue = propertyDescriptor.Value;
+                var actualFieldOption = DefaultFieldOptionds;
+                string actualFieldName = property;
                 if (_fields.TryGetValue(property, out var field) == false)
                 {
-                    field = IndexField.Create(property, DefaultFieldOptionds, DefaultFieldOptionds);
-                    _fields.Add(property, field);
+                    var shouldCreateField = true;
+                    if (actualValue.IsObject())
+                    {
+                        if (TryDetectDynamicFieldCreation(ref actualValue, property, ref actualFieldOption, out actualFieldName))
+                        {
+                            shouldCreateField = _fields.ContainsKey(actualFieldName) == false;
+                        }
+                    }
+
+                    if (shouldCreateField && actualFieldName != JavaScriptIndex.DynamicFieldName)
```

Comment:
Why is this `JavaScriptIndex.DynamicFieldName` required?

Reply:
it is required either here or on the converter and i feel this is the better spot otherwise it will show in the studio and its not a real field

Reply:
We _are_ showing it as a field in the studio for C#, and I don't want to handle this here.
Consider the case where we have _two_ such fields, etc.